src/main/java/de/uni_passau/fim/se2/pipeline_helper/checkers/line_length/LineLengthChecker.java

```diff
@@ -54,10 +46,23 @@ public CheckerResult check() throws CheckerException {
 
         StringBuilder sb = new StringBuilder();
         sb.append(String.format("Found files with lines longer than %d characters:%n", maxLength));
-        for (var entry : violations.entrySet()) {
-            sb.append(String.format("%s: %d lines%n", entry.getKey(), entry.getValue()));
-        }
+        violations.forEach(sb::append);
 
         return new CheckerResult(CHECKER_NAME, false, sb.toString().trim());
     }
+    
+    private List<Integer> getAllLineIndicesWithViolations(Path path) throws CheckerException {
+        List<Integer> lineNumbers = new ArrayList<>();
+        try {
+            final List<String> lines = Files.readAllLines(path);
+            for (int i = 0; i< lines.size(); i++) {
+                if (lines.get(i).length() > maxLength) {
+                    lineNumbers.add(i);
```

Comment:
off-by-one: `i = 0` for the first line, usually text editors count the lines starting from `1`.